### PR TITLE
Secure API routes

### LINKED ISSUE #51: Protect API routes
Check LearnDot Lab for JWT, "Securing the Routes" section

https://learn.fullstackacademy.com/workshop/5fe2759804396f0004a82f41/content/605220d83e03cf0004aa8298/text

### PR DESCRIPTION
**Pull request by tymonaghan**
created requireToken middleware and apply it to the relevant routes.

closes #51

### CODE REVIEW

File: client/store/auth.js [hunk 1]
```diff
@@ -1,60 +1,69 @@
-import axios from 'axios'
```
Comment (tymonaghan) [inline client/store/auth.js:1]: no actual changes to this file, formatting only (sorry)

File: client/store/cart.js [hunk 2]
```diff
@@ -31,7 +33,10 @@ export const getCart = (userId) => {
   // take in userId, return cart that belongTo that user
   return async (dispatch) => {
     try {
-      const axiosResponse = await axios.get(`/api/users/${userId}/cart`);
+      const token = window.localStorage.getItem(TOKEN);
```
Comment (tymonaghan) [inline client/store/cart.js:36]: NEW: get token from localStorage and pass to API route as header before allowing cart access

File: server/api/index.js [hunk 3]
```diff
@@ -1,11 +1,12 @@
-const router = require("express").Router();
+const router = require('express').Router();
 module.exports = router;
+const { requireToken } = require('../app');
 
-router.use("/users", require("./users"));
-router.use("/products", require("./products"));
+router.use('/users', requireToken, require('./users'));
```
Comment (tymonaghan) [inline server/api/index.js:5]: protect ALL /api/users routes

File: server/api/products.js [hunk 4]
```diff
@@ -27,7 +28,7 @@ router.get('/:productId', async (req, res, next) => {
 });
 
 // POST /api/products/
-router.post('/', async (req, res, next) => {
+router.post('/', requireToken, async (req, res, next) => {
```
Comment (tymonaghan) [inline server/api/products.js:31]: added requireToken middleware to relevant Product routes only

File: server/app.js [hunk 5]
```diff
@@ -1,43 +1,61 @@
...
 // logging middleware
-app.use(morgan('dev'))
+app.use(morgan('dev'));
 
 // body parsing middleware
-app.use(express.json())
+app.use(express.json());
+
+// token checking middleware (used in /api)
+async function requireToken(req, res, next) {
```
Comment (tymonaghan) [inline server/app.js:15]: define async middleware function that checks token. it is exported along with ```app``` near the top of the file

File: server/index.js [hunk 6]
```diff
@@ -1,6 +1,4 @@
-const { db } = require('./db')
-const PORT = process.env.PORT || 8080
-const app = require('./app')
+const { app } = require('./app');
```
Comment (tymonaghan) [inline server/index.js:1]: change import to match export from app.js

**Review by alstonwhite — APPROVED** [review]

**Pull request merged**